src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,94 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DEFAULT_DELIMETER = ",";
+    private static final String DATA_SUPPLY = "supply";
+    private static final String DATA_BUY = "buy";
+    private static final String DATA_RESULT = "result";
+    private static final int TYPE = 0;
+    private static final int VALUE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFromFile(fromFileName);
+        String report = processData(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private void writeToFile(String toFilename, String report) {
+        BufferedWriter bufferedWriter = null;
+        try {
+            bufferedWriter = new BufferedWriter(new FileWriter(new File(toFilename)));
+            bufferedWriter.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't open file " + toFilename, e);
+        } finally {
+            try {
+                bufferedWriter.close();
+            } catch (IOException e) {
+                throw new RuntimeException("Can't close file " + toFilename, e);
+            }
+        }
+    }
+
+    private String processData(String dataFromFile) {
+        Data supply = new Data(DATA_SUPPLY);
+        Data buy = new Data(DATA_BUY);
+        Data result = new Data(DATA_RESULT);
+        StringBuilder stringBuilder = new StringBuilder();
+        String[] lines = dataFromFile.split("\n");
+        for (String line : lines) {
+            String[] typeValue = line.split(DEFAULT_DELIMETER);
+            String type = typeValue[TYPE];
+            int value = Integer.parseInt(typeValue[VALUE]);
+            if (type.equals(DATA_SUPPLY)) {
```

Comment:
It's better to use if - else - if blocks here

Reply:
\+